### PR TITLE
Fix #293: Add Learning rate scheduler api_guide

### LINKED ISSUE #293: API Guide 第2期：learning_rate_scheduler （11月19日初稿pr，11月26日定稿merge）
# API Guide写前须知

## 背景
API Guide可以理解为是API的使用指南，旨在向用户说明API的设计思路、使用方法等
撰写计划共计约[45篇](http://agroup.baidu.com/paddlepaddle/view/office/1136638)。其中在PaddlePaddle v1.1中，已上线14篇，例如[官网-API-API使用指南-卷积](http://paddlepaddle.org/documentation/api/zh/1.1/api_guides/low_level/layers/conv.html)

## 任务说明
一篇完整的API Guide，需要包含以下几个内容：

1）对本篇主题的简介，例如：

``
卷积有两组输入：特征图和卷积核，依据输入特征和卷积核的形状、Layout不同、计算方式的不同，在Fluid里，有针对变长序列特征的一维卷积，有针对定长图像特征的二维(2D Conv)、三维卷积(3D Conv)，同时也有卷积计算的逆向过程，下面先介绍Fluid里的2D/3D卷积，再来介绍序列卷积。
``

2）对主题下每个API的介绍，例如

``
这个layer以一个mini batch的序列为输入，在每个序列内做softmax操作。其输出为一个mini batch相同shape的序列，但在序列内是经softmax归一化过的。 这个layer往往用于在每个sequence内做softmax归一化。
API Reference 请参考 sequence_softmax
``

- 完整例子：
[1.1已上线的API Guide](https://github.com/PaddlePaddle/FluidDoc/tree/release/1.1/doc/fluid/api/api_guides/low_level)

- 文档题目：请见标题

- 语言：汉语

- 格式：rst格式（因官网对rst格式支持较好，推荐用rst格式，如不了解rst，请在[这里](https://sphinx-doc-zh.readthedocs.io/en/latest/rest.html)学习）

- 字数：不限

- 效果预览：

为提高Review效率，请先在本地预览，确认格式正确再提交pr，请将预览效果截图贴在pr中，**不预览的pr不能合入**
预览工具的使用方法请见[这里](https://github.com/PaddlePaddle/FluidDoc#preview-with-ppo)

请注意：FluidDoc Repo的文档是通过Sphinx工具显示在官网上的。在Sphinx中，通过名为index.rst（或index_cn.rst等类似名称）的文件中的toctree来决定每一级显示哪几篇文档，在预览时，为了让文档能够正确预览，需要修改index，例如，如果新增一篇layers的API Guide，则需要修改`doc/fluid/api/api_guides/low_level/layers/index.rst` 这里的文件，并将自己文档的名字加入到index，如下图：

image

但在提交pr前，请不要git add 对index的修改，以避免多人同时修改这个文件引发conflict

- 提交分支：develop

- 提交位置：提交位置在`FluidDoc/doc/fluid/api/api_guides/low_level`

注：
layers文档，请提交到`FluidDoc/doc/fluid/api/api_guides/low_level/layers`
多机文档，请提交到`FluidDoc/doc/fluid/api/api_guides/low_level/cluster`

- pr提交截止日期：**请见标题**

- merge截止时间：**请见标题**

- 需要**特别注意**的格式问题

在一篇API Guide中，需要给出链接跳转至API Reference，[例如](http://paddlepaddle.org/documentation/api/zh/1.1/api_guides/low_level/layers/sequence.html#sequence-softmax)：

image

请注意：这里的超链接是用**内链**写成的，这样做的好处是，当某个API Reference的位置发生变化时，无需专门在文档里改超链接的路径，依然可以正确跳转。

用内链引用API Reference的方法： 以  ` :ref:<api_name>` 的格式引用reference（而不是其他格式：例如超链接、`xxx`等），`api_name`需与对应rst文件中的名字一致。

参考示例：https://raw.githubusercontent.com/PaddlePaddle/FluidDoc/develop_doc/doc/fluid/api_guides/low_level/optimizer/optimizer_all.rst

## 提交pr时的注意事项

- 贴预览效果

您可以选择以下两种方式展示预览效果：贴图或者给出预览网址，以方便我们检查格式

- 检查提交了几个文件

在预览工具时，需要修改index以确保文档可以显示在预览工具中，但在提交pr时，请去掉对index的修改，只提交您新写的文档，这是为了避免多人同时修改index文件引发conflict

- PR Reviewer：

为了提高Review效率，请在PR提交时增加Reviewer：一个PR的Reviewer由Peer Reviewer+ RD Reviewer+PM Reviewer组成，分别负责检查技术、格式的问题

Peer Reviewer ：1名，请找同方向的同学，例如NLP相关API请加kuke
RD Reviewer：luotao1
PM Reviewer：tink2123

- PR Labels：

为了方便管理API Guide进度，请在PR提交时增加Label `API Guide`

### PR DESCRIPTION
**Pull request by heavengate**

### CODE REVIEW

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+############
+学习率调度器
+############
+
+当我们使用诸如梯度下降法等方式来训练模型时，一般会兼顾训练速度和损失(loss)选择相对合适的学习率，但若在训练过程中一直使用一个学习率的话，训练集的损失下降到一定程度后就会不在继续下降了，而是在一定范围内震荡。其震荡原理如下图所示，即当损失函数收敛到局部极小值附近时，会由于学习率过大而导致更新步幅过大而越过极小值而出现震荡。
+
+.. image:: ../../../../images/learning_rate_scheduler.png
+    :scale: 50 %
+    :align: center
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst:11]: 图片在github里显示无比大，请缩小原始图片大小 https://github.com/PaddlePaddle/FluidDoc/blob/2d93e595bc51ba719f142315c3d3da18ab9f3b8f/doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst [hunk 2]
```diff
@@ -0,0 +1,40 @@
+.. _api_guide_learning_rate_scheduler:
+
+############
+学习率调度器
+############
+
+当我们使用诸如梯度下降法等方式来训练模型时，一般会兼顾训练速度和损失(loss)选择相对合适的学习率，但若在训练过程中一直使用一个学习率的话，训练集的损失下降到一定程度后就会不在继续下降了，而是在一定范围内震荡。其震荡原理如下图所示，即当损失函数收敛到局部极小值附近时，会由于学习率过大而导致更新步幅过大而越过极小值而出现震荡。
```
Comment (luotao1) [inline doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: - 损失(loss)来选择，增加一个“来”
- 。但若在xxx （逗号改句号）
- “由于学习率过大而导致更新步幅过大而越过极小值而出现震荡。”这句话请重新组织，一句话里面有三个“而”

Comment (heavengate) [reply doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: Done

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst (hunk 1 above)
Comment (heavengate) [reply doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst:11]: Done

**Review by heavengate — COMMENTED** [review]
Thanks for your reviewing and have done for all suggestions

**Comment by luotao1** [comment]
LGTM

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
+.. _api_guide_learning_rate_scheduler:
+
+############
+学习率调度器
+############
+
+当我们使用诸如梯度下降法等方式来训练模型时，一般会兼顾训练速度和损失(loss)来选择相对合适的学习率。但若在训练过程中一直使用一个学习率的话，训练集的损失下降到一定程度后就会不在继续下降了，而是在一定范围内震荡。其震荡原理如下图所示，即当损失函数收敛到局部极小值附近时，会由于学习率过大导致更新步幅过大，每步参数更新会反复越过极小值而出现震荡。
```
Comment (shanyi15) [inline doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: “但若在训练过程中一直使用一个学习率的话......”，这句略口语化，建议修改成更加书面语的文字，例如：但若在训练过程中一直使用一个学习率，训练集的损失下降到一定程度后便不再继续下降....

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+* :code:`exponential_decay`: 指数衰减，即每次将当前学习率乘以给定的衰减率得到下一个学习率。
+  相关API Reference请参考 :ref:`api_fluid_layers_exponential_decay`
+
+* :code:`natural_exp_decay`: 自然指数衰减，即每次将当前学习率乘以给定的衰减率的自然指数得到下一个学习率。
+  相关API Reference请参考 :ref:`api_fluid_layers_natural_exp_decay`
+
+* :code:`inverse_time_decay`: 逆时间衰减，即得到的学习率与当前衰减次数成反比。
+  相关API Reference请参考 :ref:`api_fluid_layers_inverse_time_decay`
+
+* :code:`polynomial_decay`: 多项式衰减，即得到的学习率为初始学习率和给定最终学习之间由多项式计算权重定比分点的插值
```
Comment (shanyi15) [inline doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: 插值后少了句号，请和上下文保持一致

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+当我们使用诸如梯度下降法等方式来训练模型时，一般会兼顾训练速度和损失(loss)来选择相对合适的学习率。但若在训练过程中一直使用一个学习率的话，训练集的损失下降到一定程度后就会不在继续下降了，而是在一定范围内震荡。其震荡原理如下图所示，即当损失函数收敛到局部极小值附近时，会由于学习率过大导致更新步幅过大，每步参数更新会反复越过极小值而出现震荡。
+
+.. image:: ../../../../images/learning_rate_scheduler.png
+    :scale: 80 %
+    :align: center
+
+
+学习率调度器定义了常用的学习率衰减策略来动态生成学习率，学习率衰减函数以epoch或step为参数，返回一个随训练逐渐减小的学习率，从而兼顾降低训练时间和在局部极小值能更好寻优两个方面。
+
+下面介绍学习率调度器中相关的Api。
```
Comment (shanyi15) [inline doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: 这里最后可以给一个“：”

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst (hunk 3 above)
Comment (heavengate) [reply doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: Done

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst (hunk 5 above)
Comment (heavengate) [reply doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: Done

File: doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst (hunk 4 above)
Comment (heavengate) [reply doc/fluid/api/api_guides/low_level/layers/learning_rate_scheduler.rst]: Done

**Review by heavengate — COMMENTED** [review]
Thanks for your reviewing and all have been fixed

**Review by shanyi15 — APPROVED** [review]
LGTM!
